### PR TITLE
Add doc for citus.local_table_join_policy

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
Add doc for citus.local_table_join_policy

### CODE REVIEW

File: develop/api_guc.rst [hunk 1]
```diff
@@ -222,6 +222,19 @@ The default value for this GUC is 'on'.
 Planner Configuration
 ------------------------------------------------
 
+citus.local_table_join_policy (enum)
+$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+
+Sets how citus should plan local and distributed table joins. There are four modes available:
```
Comment (onderkalaci) [inline develop/api_guc.rst:228]: should we very briefly mention why would anyone need to change this? Also, maybe move `auto` to top as that's the default?

**Review by jonels-msft — CHANGES_REQUESTED** [review]
More explanation of the terminology would be helpful. Especially the mechanics of conversion. I wasn't sure if that refers to rewriting the query in some way, or making some kind of temporary copy of tables across nodes...

File: develop/api_guc.rst [hunk 2]
```diff
@@ -222,6 +222,19 @@ The default value for this GUC is 'on'.
 Planner Configuration
 ------------------------------------------------
 
+citus.local_table_join_policy (enum)
```
Comment (jonels-msft) [inline develop/api_guc.rst:225]: We should add a link target above this, like:

```rst
.. _local_table_join_policy:
```

That way Onur can link to it from https://github.com/citusdata/citus_docs/pull/983

**Comment by jonels-msft** [comment]
@SaitTalhaNisanci thanks for the clarification! I'll make a few edits on this branch and get it merged.

**Comment by SaitTalhaNisanci** [comment]
@jonels-msft, let me know if there is anything we can clarify more because if that isn't clear to you, it will not be clear to users for sure :)

**Pull request merged**